File: _archive/tests/fixtures/sample.ts
function greet(name: string): string {
  return "Hello, " + name;
}

class UserService {
  private users: Map<string, User> = new Map();

  constructor() {
    console.log("UserService created");
  }

  addUser(user: User): void {
    this.users.set(user.id, user);
  }

  findByName(name: string): User | undefined {
    return Array.from(this.users.values()).find(u => u.name === name);
  }
}

interface User {
  id: string;
  name: string;
  email?: string;
}

enum Status {
  Active,
  Inactive,
}

type Callback = (result: string) => void;

import { Component } from "react";
import * as fs from "fs";

File: _archive/tests/fixtures/sample.rs
use std::collections::HashMap;
use crate::utils::*;

/// A greeter struct
pub struct Greeter {
    name: String,
}

impl Greeter {
    pub fn new(name: String) -> Self {
        Greeter { name }
    }

    pub fn greet(&self) -> String {
        format!("Hello, {}!", self.name)
    }
}

/// User status enum
pub enum Status {
    Active,
    Inactive,
}

/// User trait
pub trait User {
    fn id(&self) -> u64;
}

/// Admin struct implementing User
pub struct Admin {
    id: u64,
}

impl User for Admin {
    fn id(&self) -> u64 {
        self.id
    }
}

pub type Callback = Box<dyn Fn(String)>;

const MAX_RETRIES: u32 = 3;

pub fn greet(name: &str) -> String {
    format!("Hello, {}!", name)
}
